addons/mail/models/models.py

```diff
@@ -31,3 +34,123 @@ def _notify_email_header_dict(self):
         return {
             'X-Odoo-Objects': "%s-%s" % (self._name, self.id),
         }
+
+    @api.model
+    def _notify_admins(self, subject, body, doc=False, **kwargs):
+        """
+            Posts notifications about system failures on Admins channel (recreates the channel if it no longer exists).
+        """
+        ir_model_data = self.env['ir.model.data']
+        model_mail_channel = self.env['mail.channel']
+        odoobot_id = ir_model_data.xmlid_to_res_id("base.partner_root")
+        # Get channel
+        admins_channel = ir_model_data.xmlid_to_object('mail.channel_admins') or model_mail_channel.sudo().search([('name', '=', str('Administrators'))])
+        # Create a new channel if it no longer exists
+        if not admins_channel:
+            admins_channel = model_mail_channel.sudo().with_context(mail_create_nosubscribe=True).create({
+                'group_ids': [(4, self.env.ref('base.group_system').id)],
+                'group_public_id': self.env.ref('base.group_system').id,
+                'public': 'groups',
+                'channel_type': 'channel',
+                'email_send': False,
+                'name': 'Administrators',
+                'description': _('Channel to post notifications about system silent failures to make admins aware of them.')
+            })
+        # Get the message to post (body) from template
+        message = self._format_notification(subject, body, doc)
+        # Check Repeat delay (default : 7 days -> can be updated in method parameters)
+        channel_last_message = self.env['mail.message'].search(['&', ("channel_ids", "in", [admins_channel.id]), ("subject", "=", subject)], limit=1)
+        channel_last_message_valide = False
+        if channel_last_message:
+            repeat_delay = kwargs.get('repeat_delay', None) or timedelta(days=7)
+            repeat_last_datetime = datetime.now() - repeat_delay
+            channel_last_message_valide = channel_last_message.date < repeat_last_datetime
+
+        if not channel_last_message or channel_last_message_valide:
+            admins_channel.sudo().message_post(subject=subject, body=message, author_id=odoobot_id, message_type="comment", subtype_xmlid="mail.mt_comment")
+
+    @api.model
+    def _format_record_link(self):
+        """
+            Get record local url (used to get links to targeted records in notifications : mail servers with failures...).
+        """
+        return ('<a target="_blank" href="/web#id={0}&view_type=form&model={1}">{2}</a>').format(self.id, self._name, self.name)
```

Comment:
Please use the ``/mail/view`` route.

Example: ``/mail/view?model=%s&res_id=%s``

This allows to redirect to the backend/frontend according to your access rights. And it less subject to be broken ;)